node-fs-readfile/cat.js

```diff
@@ -3,16 +3,17 @@ import { readFile } from 'node:fs/promises';
 const fileArray = process.argv.slice(2);
 
 async function print() {
-  await Promise.all(
-    fileArray.map(async (file) => {
-      try {
-        const data = await readFile(`./${file}`, { encoding: 'utf8' });
-        console.log(data);
-      } catch (error) {
-        console.log(error.message);
-      }
-    })
-  );
+  const array = await fileArray.map((file) => {
+    try {
+      const read = readFile(file, { encoding: 'utf8' });
+      return read;
+    } catch (error) {
+      return error;
+    }
```

Comment:
Good, this fixes the order problem. Note, though, that it's doing more than it needs to. It's usually easier to just create the promises array with just simple `readFile` calls and then put the try/catch around Promise.all.